packages/realm/src/OrderedCollection.ts

```diff
@@ -132,21 +135,24 @@ export abstract class OrderedCollection<T = unknown, EntryType extends [unknown,
     });
     // Wrap in a proxy to trap ownKeys and get, enabling the spread operator
     const proxied = new Proxy(this, PROXY_HANDLER as ProxyHandler<this>);
+    const itemType = internal.type; // Can probably get this passed in rather than dropping to C++.
+    const isObjectType =
+      (itemType & ~binding.PropertyType.Flags) === binding.PropertyType.Object ||
+      (itemType & ~binding.PropertyType.Flags) === binding.PropertyType.LinkingObjects;
     // Get the class helpers for later use, if available
-    const { objectType } = results;
-    const classHelpers = typeof objectType === "string" && objectType !== "" ? realm.getClassHelpers(objectType) : null;
+    const classHelpers = isObjectType && objectSchemaName ? realm.getClassHelpers(objectSchemaName) : null;
     // Make the internal properties non-enumerable
     Object.defineProperty(this, "realm", {
```

Comment:
There is a large perf boost at least on node from not using `Object.defineProperty()` on these. Is it actually required? If if it is, it is likely faster to define a single property which is a plain object with all of these fields.